### PR TITLE
Lianalite login

### PR DESCRIPTION
**Pull request by edouardparis**
This PR introduce the new login to the remote backend.

The choice of the used backend is defined per wallet.
The authentication credentials for the remote backend are stored in the settings.json file.

## In the installer

The choice between two different backends is  showned to the user that can choose to authenticate to the remote backend or use the local bitcoind.

## Between the launcher and the app a new temporary panel: login.

This intermediary state after the choice of network datadir from the user, is used to connect to the remote backend if the wallet information in the settings.json file contains the credentials for the remote backend authentication otherwise the launcher redirects directly to the App state with the embedded daemon running.

### CODE REVIEW

**Review by jp1ac4 — COMMENTED** [review]
Some initial comments about wording.

I'll continue with testing.

**Comment by pythcoiner** [comment]
Maybe we should make the token more visible in the login email:
image
(line return + centered + highlight)

**Comment by pythcoiner** [comment]
here the backgroud color of the buttons don't change on hover:
image

**Comment by pythcoiner** [comment]
here maybe we should no display "lianad v" if we connect on remote backend:
image

**Comment by pythcoiner** [comment]
i got this once, same issue w/ all mail addresses (w/ or w/o already created wallet on theses addresse), i cannot reproduce after restart liana:
image

**Comment by edouardparis** [comment]
> it happen if i try creating a mainnet wallet

It is because LianaLite mainnet website is not yet running. GUI is looking at lianalite.com/api/env to get to which backend it should communicate with.

**Comment by pythcoiner** [comment]
> Lite mainnet website is not

will LianaLite backend run before v7 release?

**Comment by edouardparis** [comment]
> will LianaLite backend run before v7 release?

Yes, this is the goal

File: gui/src/lianalite/login.rs [hunk 1]
```diff
@@ -0,0 +1,564 @@
...
+                    connection_error: None,
+                    wallet_id: auth_config.wallet_id.clone(),
+                    auth_error: None,
+                    processing: true,
+                },
+                Command::perform(
+                    async move {
+                        let service_config = super::client::get_service_config(network)
+                            .await
+                            .map_err(|e| Error::Unexpected(e.to_string()))?;
```
Comment (pythcoiner) [inline gui/src/lianalite/login.rs:156]: shouldn't we give a clearer error here? i think it's where we 404 atm on mainnet?

File: gui/src/lianalite/login.rs [hunk 2]
```diff
@@ -0,0 +1,564 @@
...
+                        )
+                        .await
+                    },
+                    Message::Connected,
+                ),
+            ),
+        }
+    }
+
+    pub fn stop(&mut self) {}
```
Comment (pythcoiner) [inline gui/src/lianalite/login.rs:176]: futur use or placeholder?

File: gui/src/lianalite/login.rs [hunk 3]
```diff
@@ -0,0 +1,564 @@
...
+                        let email = email.value.clone();
+                        let network = self.network;
+                        self.processing = true;
+                        self.connection_error = None;
+                        self.auth_error = None;
+                        return Command::perform(
+                            async move {
+                                let config = super::client::get_service_config(network)
+                                    .await
+                                    .map_err(|e| Error::Unexpected(e.to_string()))?;
```
Comment (pythcoiner) [inline gui/src/lianalite/login.rs:219]: shouldn't we give a clearer error here? i think it's where we 404 atm on mainnet?

File: gui/src/lianalite/login.rs [hunk 4]
```diff
@@ -0,0 +1,564 @@
...
+                                            )
+                                            .push(button::primary(None, "Next").on_press_maybe(
+                                                if self.processing {
+                                                    None
+                                                } else {
+                                                    Some(ViewMessage::RequestOTP)
+                                                },
+                                            )),
+                                        ConnectionStep::EnterOtp { otp, .. } => Column::new()
+                                            .push(text("An authentication was send to you mail"))
```
Comment (pythcoiner) [inline gui/src/lianalite/login.rs:397]: nit: 'to your email address' ?

File: gui/src/installer/step/backend.rs [hunk 5]
```diff
@@ -0,0 +1,541 @@
...
+            } = &mut self.step
+            {
+                *remote_backend_is_selected = false;
+            }
+            return Command::perform(async move {}, |_| Message::Next);
+        }
+        match &mut self.step {
+            ConnectionStep::EnterEmail { email } => match message {
+                Message::SelectBackend(message::SelectBackend::EmailEdited(value)) => {
+                    email.valid = value.is_empty() || email_address::EmailAddress::is_valid(&value);
```
[the remark this answers is not in the document]
Comment (edouardparis) [reply gui/src/installer/step/backend.rs:91]: Better indeed !

File: gui/src/installer/view.rs [hunk 6]
```diff
@@ -374,8 +376,197 @@ pub fn recovery_path_view(
...
+        },
+    );
+
+    layout(
+        progress,
+        email,
+        "Add wallet",
+        Column::new()
+            .spacing(50)
+            .push_maybe(error.map(|e| card::error("Something wrong happened", e.to_string())))
```
[the remark this answers is not in the document]
Comment (edouardparis) [reply gui/src/installer/view.rs:558]: card::error is a component that have the error title displayed in a disclaimer and the message displayed on hover

**Review by edouardparis — COMMENTED** [review]
Self-ACK 80077905a64b2d804c20a49d190f742a244d7816
This PR is quite in a good state, maybe some UI change may be needed but they will tackled in next iteration from the team feedbacks.

**Pull request merged**